politeiad/cache/cockroachdb/models.go

```diff
@@ -47,13 +47,14 @@ func (MetadataStream) TableName() string {
 
 // Record is an entire record and it's content.
 type Record struct {
-	Key       string `gorm:"primary_key"`       // Primary key (token+version)
-	Token     string `gorm:"not null;size:64"`  // Censorship token
-	Version   uint64 `gorm:"not null"`          // Version of files
-	Status    int    `gorm:"not null"`          // Current status
-	Timestamp int64  `gorm:"not null"`          // UNIX timestamp of last updated
-	Merkle    string `gorm:"not null;size:64"`  // Merkle root of all files in record
-	Signature string `gorm:"not null;size:128"` // Server signature of merkle+token
+	Key         string `gorm:"primary_key"`       // Primary key (token+version)
+	Token       string `gorm:"not null;size:64"`  // Censorship token
+	Version     uint64 `gorm:"not null"`          // Version of files
+	Status      int    `gorm:"not null"`          // Current status
+	Timestamp   int64  `gorm:"not null"`          // UNIX timestamp of last updated
+	Merkle      string `gorm:"not null;size:64"`  // Merkle root of all files in record
+	Signature   string `gorm:"not null;size:128"` // Server signature of merkle+token
+	TokenPrefix string `gorm:"not null;size:8"`   // Prefix of token used for lookups
```

Comment:
If we're relying on the token prefix for lookups, should this be indexed?  Also, does the version need to be included?

Reply:
It could be indexed, but that's really just another change outside the scope of this one. There was no index on columns even before this.
What do you mean by including the version? I actually updated the primary key to be prevfix+version, but I missed updating the comment here.

Reply:
There was no index on the token column before because that's not how proposals are looked up.  They're lookup up using the token+version, which was the primary key so there was no need for an index.  

We need to support lookups for both the prefixed token and the full token.  I would keeps the primary key the same and just create a TokenPrefix index so that both lookups are just as fast.

Reply:
The lookup can still be done with the full token if the primary key is prefix+version. The token just has to be converted to its prefix.

Reply:
Lookups of the latest version of records are currently being done using the Token column. In the PR I've updated this to use the TokenPrefix column. An index does make sense on the TokenPrefix column to speed up these lookups.

Since there is a one to one mapping between Token and Prefix, and it very easy to get a prefix from a token (but not the other way around), I think lookups should always be done on the prefix. When getting a specific version, the primary key (Prefix+Version) should be used, and when getting the lastest version, the TokenPrefix column should be used.

Reply:
Throughout the codebase, it's much more common for proposals to be referenced by their full token so I'd leave the primary key as the token+version.  Changing it would be a pain to update.  The only time a proposal needs to be referenced by its prefix is for the short proposal details route.

Reply:
Ok sounds good, that makes sense.